android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/MessageCenter.java

```diff
@@ -0,0 +1,80 @@
+package com.shopify.unity.buy;
+
+import android.os.Handler;
+import android.os.HandlerThread;
+
+import com.unity3d.player.UnityPlayer;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MessageCenter {
+    private static final HandlerThread handlerThread = new HandlerThread("UnityMessageThread");
+    private static final Handler messageHandler = new Handler(handlerThread.getLooper());
+
+    private static Map<String, WeakReference<MessageCallbacks>> callbacksInWaiting = new HashMap<>();
```

Comment:
I thought so but wasn't sure. I'll remove the weak map in favor of the weakified fragment ref.